NIRCam Aperture Photometry Notebook

mcorrenti opened the pull request.

rosteen commented:
**Technical review comments**

Thanks for the submission, @mcorrenti . Overall the code in the notebook is looking very good, but the non-code documentation and such around it needs to be expanded/elaborated. The notebook runs well/quickly, and nothing big stood out to me in terms of code style problems. Some specific things that I saw as needing improvement/fixing:

- The line to download data from Box is commented out in the committed version, which uses a local file. That will need to be changed before merging, as it causes the cell to fail.

- In general, I think the biggest update needed is additional documentation - I don't think that someone new to aperture photometry using this notebook as a learning guide would have a great idea of what exactly the code is doing and why. Essentially I think you should knock the "assumed reader knowledge" down a level. "Aperture Photometry" section 2 is the most important section to add some more explanation to, I think, since that's the main point of the notebook, and some inline comments in the large code blocks would be nice as well.

- Related to the above (assumed reader knowledge), it would be nice to explain what the detector/filter constants being set are (psf_fwhm, zp-modA, etc), and link to the location they are sourced from.

- The notebook needs the STScI footer and "About this Notebook" section at the end, see [the style guide](https://github.com/spacetelescope/style-guides/blob/master/guides/jupyter-notebooks.md#footer)

- In the print statement "Conversion factor from {units} to cps for filter {f}:" it's not immediately clear to me what "cps" is in this context (counts per second?) and why the conversion needs to be done. You could address this by adding some more detail to the "Finding Sources" markdown section.

- Super minor detail, but "var_bkg" is getting set to a constant value inside a loop - you could pull that out to the top of the cell, where it would be right below where it's mentioned in the Markdown as something the user can change.

mcorrenti commented:
Thanks for your review @rosteen. Below my comments/replies:

- Removed comments that prevent to download the files from BOX. Included an "if" statement that allows to skip the download if the files exist in the current folder.
- Included a paragraph to provide references to the photometric zeropoints and PSF FWHM.
- Included a brief explanation of the daofind algorithm and added reference to Stetson (1987) paper.
- Included a paragraph to explain the transformation from MJy/sr (surface brightness, original output from the pipeline) to DN/s.
- Moved the finding sources algorithm inside a function.
- Included the STScI footer and About this notebook section.
- Moved the "var_bkg" outside of the loop and included as a parameter in the finding source function.
- Verified PEP-8 compliance running flake8-nb on the notebook

rosteen reviewed (approved):
Thanks for addressing my comments. I confirmed that the notebook still runs successfully, approving.
